src/problems/basic_problems.jl

```diff
@@ -299,15 +299,30 @@ Any extra keyword arguments are captured to be sent to the optimizers.
 
 ### Fields
 
-* `f`: The function in the problem.
-* `u0`: The initial guess for the optima.
-* `p`: The parameters for the problem. Defaults to `NullParameters`.
+* `f`: the function in the problem.
+* `u0`: the initial guess for the optima.
+* `p`: the parameters for the problem. Defaults to `NullParameters`.
 * `lb`: the lower bounds for the optimization of `u`.
 * `ub`: the upper bounds for the optimization of `u`.
-* `lcons`:
-* `ucons`:
-* `sense`:
-* `kwargs`: The keyword arguments passed on to the solvers.
+* `lcons`: the vector of lower bounds for the constraints passed to [OptimizationFunction](@ref).
+    Defaults to `nothing`, implying no lower bounds for the constraints (i.e. the constraint bound is `-Inf`)
+* `ucons`: the vector of upper bounds for the constraints passed to [OptimizationFunction](@ref).
+    Defaults to `nothing`, implying no upper bounds for the constraints (i.e. the constraint bound is `Inf`)
+* `sense`: the objective sense, can take `MaxSense` or `MinSense` from Optimization.jl.
+* `kwargs`: the keyword arguments passed on to the solvers.
+
+## Inequality and Equality Constraints
+
+Both inequality and equality constraints are defined by the `f.cons` function in the `OptimizationFunction`
+description of the problem structure. This `f.cons` is given as a function `f.cons(u,p)` which computes
+the value at `u` of the constraints. For example, take `f.cons(u,p) = u[1] - u[2]` 
```

Comment:
```suggestion
the value of the constraints at `u`. For example, take `f.cons(u,p) = u[1] - u[2]`.
```

Reply:
In-place v.s. out-of-place is still a little confusing to me, but perhaps that's because of the state of the rest of the optimization docs. If it's not addressed by any other issues I would suggest https://github.com/SciML/Optimization.jl/issues/312 be added to the roadmap